src/main/java/core/basesyntax/Program1.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Program1 {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < figures.length; i++) {
+            if (i <= figures.length / 2) {
```

Comment:
The condition `i <= figures.length / 2` will result in more than half of the figures being random because the index starts at 0. To ensure exactly half are random, use `i < figures.length / 2`.